### PR TITLE
Support joinCommentLines setting with experimental formatter

### LINKED ISSUE #1243: Support `xml.format.joinCommentLines` setting with experimental formatter
See https://github.com/redhat-developer/vscode-xml/blob/master/docs/Formatting.md#xmlformatjoincommentlines

### PR DESCRIPTION
**Pull request by JessicaJHee**
Fixes #1243 

Signed-off-by: Jessica He <jhe@redhat.com>

### CODE REVIEW

**Comment by JessicaJHee** [comment]
This PR only considers the tests/use cases for the joinCommentLines setting. There is a lot more missing formatting for comments that are unrelated to this setting that could be done in another PR.

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMCommentFormatter.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+import java.util.List;
+
+import org.eclipse.lemminx.dom.DOMComment;
+import org.eclipse.lsp4j.TextEdit;
+
+/**
+ * DOM text formatter.
+ * 
+ * @author Angelo ZERR
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMCommentFormatter.java:22]: Please update the author

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMCommentFormatter.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+*******************************************************************************/
+package org.eclipse.lemminx.services.format;
+
+import java.util.List;
+
+import org.eclipse.lemminx.dom.DOMComment;
+import org.eclipse.lsp4j.TextEdit;
+
+/**
+ * DOM text formatter.
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMCommentFormatter.java:20]: DOM comment formatter

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMCommentFormatter.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	public void formatComment(DOMComment commentNode, XMLFormattingConstraints parentConstraints,
+			List<TextEdit> edits) {
+
+		// Check the comment is closed properly
+		if (commentNode.getEnd() == commentNode.getEndContent()) {
+			return;
+		}
+		FormatElementCategory formatElementCategory = parentConstraints.getFormatElementCategory();
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMCommentFormatter.java:40]: I think comment text should not be formatted acording the FormatElementCategory

Please remove the following  ode which formats comment text according FormatElementCategory

**Comment by angelozerr** [comment]
> There is a lot more missing formatting for comments that are unrelated to this setting that could be done in another PR.

Indeed let support comment formatting step by step with several p. It will easier to review it. Thats why please remove the code with formatelementcategory which is used for text node and nor for text comments node.

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/experimental/XMLFormatterJoinCommentLinesTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+				"  <!-- commentcommentcommentcomment commentcommentcommentcomment" + lineSeparator() + //
+				"  commentcommentcommentcomment commentcommmentcommentcommentcomment" + lineSeparator() + //
+				"  commentcommentcommentscommentcomment commentcommentcommentscommentcomment" + lineSeparator() + //
+				"  commentcommentcomments --></a>";
+		SharedSettings settings = new SharedSettings();
+		settings.getFormattingSettings().setJoinCommentLines(true);
+		assertFormat(content, expected, settings);
+	}
+
+	private static void assertFormat(String unformatted, String expected, SharedSettings sharedSettings)
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/experimental/XMLFormatterJoinCommentLinesTest.java:120]: Remove this method, with experimental formatter, we need to test the text edit (if text edit is empty or not).

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/experimental/XMLFormatterJoinCommentLinesTest.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+		String content = "<!--" + lineSeparator() + //
+				" line 1" + lineSeparator() + //
+				" " + lineSeparator() + //
+				" " + lineSeparator() + //
+				"   line 2" + lineSeparator() + //
+				" -->";
+		String expected = "<!-- line 1 line 2 -->";
+		SharedSettings settings = new SharedSettings();
+		settings.getFormattingSettings().setJoinCommentLines(true);
+		assertFormat(content, expected, settings);
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/experimental/XMLFormatterJoinCommentLinesTest.java:38]: Declare expected text edit with te(...).

Comment (angelozerr) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/format/experimental/XMLFormatterJoinCommentLinesTest.java:38]: End your test with 

`assertFormat(expected, expected, settings);`

to check that it will not generate some textedit if teh XML is already formatted. Do that for the other tests.
